x-pack/legacy/plugins/apm/server/lib/service_map/initialize_service_maps.ts

```diff
@@ -0,0 +1,109 @@
+  const config = server.config();
+
+  setupRequiredScripts(server).catch(error => {
+    server.log(
+      'error',
+      'Unable to set up required scripts for APM Service Maps:'
+    );
+    server.log('error', error);
+  });
+
+  const taskManager = server.plugins.task_manager;
+  if (taskManager) {
+    // @ts-ignore
+    const kbnServer = this.kbnServer;
+    taskManager.registerTaskDefinitions({
+      // serviceMap is the task type, and must be unique across the entire system
+      serviceMap: {
+        // Human friendly name, used to represent this task in logs, UI, etc
+        title: 'ServiceMapTask',
+        type: 'serviceMap',
+
+        // Optional, human-friendly, more detailed description
+        description: 'Extract connections in traces for service maps',
+
+        // Optional, how long, in minutes, the system should wait before
+        // a running instance of this task is considered to be timed out.
+        // This defaults to 5 minutes.
+        timeout: '5m',
+
+        // The serviceMap task occupies 2 workers, so if the system has 10 worker slots,
+        // 5 serviceMap tasks could run concurrently per Kibana instance. This value is
+        // overridden by the `override_num_workers` config value, if specified.
+        // numWorkers: 1,
+
+        // The createTaskRunner function / method returns an object that is responsible for
+        // performing the work of the task. context: { taskInstance, kbnServer }, is documented below.
+        createTaskRunner({ taskInstance }: RunContext) {
+          // Perform the work of the task. The return value should fit the TaskResult interface, documented
+          // below. Invalid return values will result in a logged warning.
+          return {
+            async run() {
+              const { state } = taskInstance;
+
+              const { mostRecent } = await runServiceMapTask(
+                kbnServer,
+                config,
+                state.lastRun
+              );
+              // console.log(`Task run count: ${(state.count || 0) + 1}`);
+
+              return {
+                state: {
+                  count: (state.count || 0) + 1,
+                  lastRun: mostRecent
+                }
+              };
+            }
+          };
+        }
+      }
+    });
+
+    // @ts-ignore
+    this.kbnServer.afterPluginsInit(async () => {
+      const fetchedTasks = await taskManager.fetch({
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  _id: 'task:servicemap-processor'
+                }
+              },
+              {
+                term: {
+                  'task.taskType': 'serviceMap'
+                }
+              }
+            ]
+          }
+        }
+      });
+      if (fetchedTasks.docs.length) {
+        await taskManager.remove('servicemap-processor');
+      }
+      await taskManager.schedule({
```

Comment:
I'm not sure if the point of this is to prevent this task being scheduled more than once, but if that is the case, then you can now use `ensureScheduled`: https://github.com/elastic/kibana/pull/50232